app/uk/gov/hmrc/fileupload/controllers/EnvelopeChecker.scala

```diff
@@ -20,46 +20,90 @@ import cats.data.Xor
 import play.api.Logger
 import play.api.http.Status._
 import play.api.libs.iteratee.Done
-import play.api.libs.json.Json
+import play.api.libs.json.{JsValue, Json}
 import play.api.libs.streams.Accumulator
 import play.api.mvc.Results._
 import play.api.mvc.{EssentialAction, RequestHeader, Result}
 import uk.gov.hmrc.fileupload.EnvelopeId
-import uk.gov.hmrc.fileupload.transfer.TransferService.{EnvelopeStatusNotFoundError, _}
+import uk.gov.hmrc.fileupload.quarantine.Constraints
+import uk.gov.hmrc.fileupload.transfer.TransferService._
 import uk.gov.hmrc.fileupload.utils.StreamsConverter
 
 import scala.concurrent.{ExecutionContext, Future}
 
 object EnvelopeChecker {
 
-  type WithValidEnvelope = EnvelopeId => EssentialAction => EssentialAction
+  type FileSize = Int
+  type WithValidEnvelope = EnvelopeId => (FileSize => EssentialAction) => EssentialAction
 
   import uk.gov.hmrc.fileupload.utils.StreamImplicits.materializer
 
-  def withValidEnvelope(check: (EnvelopeId) => Future[EnvelopeStatusResult])
+  val defaultFileSize = 10 * 1024 * 1024
+
+  def withValidEnvelope(checkEnvelopeDetails: (EnvelopeId) => Future[EnvelopeDetailResult])
                        (envelopeId: EnvelopeId)
-                       (action: EssentialAction)
+                       (action: FileSize => EssentialAction)
                        (implicit ec: ExecutionContext) =
     EssentialAction { implicit rh =>
       Accumulator.flatten {
-        check(envelopeId).map {
-          case Xor.Right("OPEN") =>
-            action(rh)
-          case Xor.Right(otherStatus) =>
-            logAndReturn(LOCKED, s"Unable to upload to envelope: $envelopeId with status: $otherStatus")
-          case Xor.Left(EnvelopeStatusNotFoundError(_)) =>
+        checkEnvelopeDetails(envelopeId).map {
+          case Xor.Right(envelope) =>
+            val status = (envelope \ "status").as[String]
+            if (status == "OPEN") {
+              action(setMaxFileSize(envelope))(rh)
+            } else if (status == "CLOSED") {
+              logAndReturn(LOCKED, s"Unable to upload to envelope: $envelopeId with status: $status")
+            } else if(status == "SEALED") {
+              logAndReturn(LOCKED, s"Unable to upload to envelope: $envelopeId with status: $status")
+            } else {
+              //If uploading to deleted Envelope
+              logAndReturn(BAD_REQUEST, s"Unable to upload to envelope: $envelopeId with status: $status")
+            }
+          case Xor.Left(EnvelopeDetailNotFoundError(_)) =>
             logAndReturn(NOT_FOUND, s"Unable to upload to nonexistent envelope: $envelopeId")
           case Xor.Left(error) =>
             logAndReturn(INTERNAL_SERVER_ERROR, error.toString)
         }
       }
     }
 
+  def setMaxFileSize(envelope: JsValue): FileSize = {
```

Comment:
wrong file naming, this method retrieves allowed max file size limit from envelope Json, does not set it.